packages/ui/src/components/pagination/components/pagination-total-pages/pagination-total-pages.component.tsx

```diff
@@ -0,0 +1,233 @@
+
+  const generateHandleOnClickBackwards = useCallback(
+    (
+      current: number,
+      infinite: boolean,
+      forwardOnly: boolean,
+      onChange: (page: number) => unknown,
+      totalPages: number,
+    ) =>
+      () => {
+        if (infinite && forwardOnly) {
+          return onChange(totalPages);
+        }
+        if (!forwardOnly) {
+          return onChange(current - 1);
+        }
+      },
+    [],
+  );
+
+  const paginationBackProps = useMemo(() => {
+    const fowardOnly = (current || 1) <= 1;
+    const defaultBackProps = {
+      disabled: !infinite && fowardOnly,
+      firstItem: true,
+      'aria-label': backAriaLabel,
+    };
+    if (defaultBackProps.disabled) {
+      return {
+        ...defaultBackProps,
+        tag: 'button',
+      } as const;
+    }
+
+    if (onChange) {
+      return {
+        ...defaultBackProps,
+        tag: linkComponent || 'button',
+        ...(current && {
+          onClick: generateHandleOnClickBackwards(current, infinite, fowardOnly, onChange, totalPages),
+        }),
+      };
+    }
+    return {
+      ...defaultBackProps,
+      tag: linkComponent || 'a',
+      href: fowardOnly ? onPageItemProps(totalPages).href : onPageItemProps(current - 1)?.href,
+    };
+  }, [
+    current,
+    infinite,
+    backAriaLabel,
+    onChange,
+    linkComponent,
+    onPageItemProps,
+    totalPages,
+    generateHandleOnClickBackwards,
+  ]);
+
+  const generateHandleOnClickForward = useCallback(
+    (current: number, infinite: boolean, backwardsOnly: boolean, onChange: (page: number) => unknown) => () => {
+      if (infinite && backwardsOnly) {
+        return onChange(1);
+      }
+      if (backwardsOnly) {
+        return;
+      }
+      return onChange(current + 1);
+    },
+    [],
+  );
+
+  const paginationNextProps = useMemo(() => {
+    const backwardsOnly = (current || 1) >= totalPages;
+    const defaultNextProps = {
+      disabled: !infinite && backwardsOnly,
+      tag: linkComponent,
+      lastItem: true,
+      'aria-label': nextAriaLabel,
+    };
+    if (defaultNextProps.disabled) {
+      return {
+        ...defaultNextProps,
+        tag: 'button',
+      } as const;
+    }
+    if (onChange) {
+      return {
+        ...defaultNextProps,
+        tag: linkComponent || 'button',
+        ...(current && {
+          onClick: generateHandleOnClickForward(current, infinite, backwardsOnly, onChange),
+        }),
+      };
+    }
+    return {
+      ...defaultNextProps,
+      tag: linkComponent || 'a',
+      href: backwardsOnly ? onPageItemProps(1)?.href : onPageItemProps(current || 0)?.href,
+    };
+  }, [
+    current,
+    totalPages,
+    infinite,
+    linkComponent,
+    nextAriaLabel,
+    onChange,
+    onPageItemProps,
+    generateHandleOnClickForward,
+  ]);
+
+  /** Compute which pages and ellipses to show */
+  const pagesToRender = useMemo(() => {
+    const minEdgePagesVisible = finalSiblingCount * 2 + finalBoundaryCount + 2;
```

Comment:
Need to adjust the calculations for total pages < 7. Is causing overflow errors
<img width="896" height="141" alt="Screenshot 2025-07-11 at 10 15 12 am" src="https://github.com/user-attachments/assets/4c05f8ca-eddb-43f7-b20a-2ffa936fd145" />